microplan-init.js

```diff
@@ -2,52 +2,86 @@ var program = require('commander')
 var fs = require('fs-extra')
 var path = require('path')
 var utils = require('./utils/utils.js')
+var async = require('async')
+var readline = require('readline')
 
 var templateFileLocation = {
   'yaml': 'parsers/yaml/template.yml',
   'json': 'parsers/json/template.json'
 }
 
 var suppFileExtns = ['.json', '.yml', '.yaml']
+var userOptsYes = ['y', 'yes', 'Y', 'YES', 'Yes']
 
-var _copyTemplateToPath = function (templateFilePath, destFilePath) {
-  let sourcePath = ''
-  try {
-    fs.statSync(templateFilePath)
-    sourcePath = templateFilePath
-  } catch (err) {
-    sourcePath = path.join(__dirname, templateFilePath)
-  }
-  initArgs.forEach(function (initArg) {
-    try {
-      fs.copySync(sourcePath, destFilePath)
-    } catch (err) {
+var rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+})
+
+async.waterfall([
+  _parseInputFromUser,
+  _getOptsFromUser,
+  _copyTemplateFile
+],
+  function (err, result) {
+    // result now equals 'done'
+    if (err) {
       console.error(err)
+      process.exit(1)
     }
-    console.log('init: %s', destFilePath)
+    console.log(result)
+    process.exit(0)
   })
-}
-program
-  .option('-t, --template <location>', 'specify input template location', templateFileLocation.yaml)
-  .parse(process.argv)
 
-var initArgs = program.args
+function _parseInputFromUser (callback) {
+  program
+    .option('-t, --template <location>', 'specify input template location', templateFileLocation.yaml)
+    .parse(process.argv)
 
-if (initArgs.length > 1) {
-  console.error('Filename required. Please specify a valid', suppFileExtns.join('/'), 'template location as argument.')
-  process.exit(1)
-}
+  var initArgs = program.args
 
-if (utils.fileExists(program.template) === false) {
-  console.error('Please specify a valid', suppFileExtns.join('/'), 'template file location.')
-  process.exit(1)
-}
+  if (initArgs.length < 1) {
+    console.error('Filename required. Please specify a valid', suppFileExtns.join('/'), 'template location as argument.')
+    process.exit(1)
```

Comment:
instead of logging error here and exiting, let's use the callback to report like we do here ( https://github.com/argonlaser/microplan/blob/master/microplan-publish.js#L59-L62 )